sw/device/lib/testing/test_status.c

```diff
@@ -0,0 +1,31 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/testing/test_status.h"
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/log.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/runtime/hart.h"
+
+/**
+ * Address of the memory location to write the test status. For DV use only.
+ */
+static const uintptr_t kSwDvTestStatusAddr = 0x1000fff8;
+
+void set_test_status(const test_status_t test_status) {
+  if (kDeviceType == kDeviceSimDV) {
+    mmio_region_t sw_dv_test_status_addr =
+        mmio_region_from_addr(kSwDvTestStatusAddr);
+    mmio_region_write32(sw_dv_test_status_addr, 0x0, (uint32_t)test_status);
+  }
+  if (test_status == kTestStatusPassed) {
+    LOG_INFO("PASS");
+    abort();
+  }
+  if (test_status == kTestStatusFailed) {
+    LOG_INFO("FAIL");
+    abort();
+  }
```

Comment:
Optional: use a `switch` statement to make this more concise.

Reply:
Done.